src/Reflex/Dom/Builder/Static.hs

```diff
@@ -0,0 +1,168 @@
+  {-# INLINABLE newEventWithLazyTriggerWithOnComplete #-}
+  newEventWithLazyTriggerWithOnComplete _ = return never
+
+type SupportsStaticDomBuilder t m = (Reflex t, MonadIO m, MonadHold t m, MonadFix m, PerformEvent t m, Performable m ~ m, MonadReflexCreateTrigger t m, Deletable t m, MonadRef m, Ref m ~ Ref IO)
+
+data StaticDomSpace
+instance DomSpace StaticDomSpace where
+    type RawElement StaticDomSpace = ByteString
+
+instance SupportsStaticDomBuilder t m => DomBuilder t (StaticDomBuilderT t m) where
+  type DomBuilderSpace (StaticDomBuilderT t m) = StaticDomSpace
+  {-# INLINABLE textNode #-}
+  textNode (TextNodeConfig contents) = StaticDomBuilderT $ do
+    modify . (:) $ constant $ BL.toStrict $ toLazyByteString $ fromHtmlEscapedText contents
+    return TextNode
+
+  {-# INLINABLE element #-}
+  element elementTag cfg child = do
+    let toAttr (_mns,k) v = let key = encodeUtf8 k; val = BL.toStrict $ toLazyByteString $ fromHtmlEscapedText v in [key <> "=\"" <> val  <> "\""]
+    let attrs0 = Map.foldMapWithKey toAttr $ cfg ^. initialAttributes
+    let attrs1 = if null attrs0 then "" else " " <> B8.intercalate " " attrs0
+
+    let tag' = encodeUtf8 elementTag
+    let open' more = constant ("<" <> tag' <> attrs1 <> " ") <> more <> constant ">"
+    let close' = constant $ "</" <> tag' <> ">"
+
+    es <- newFanEventWithTrigger $ \_ _ -> return (return ())
+    StaticDomBuilderT $ do
+      (result, innerHtml) <- lift $ runStaticDomBuilderT child
+      initial <- lift $ performEvent $ ffor (cfg ^. modifyAttributes) $ ifoldrM (\r@(mAttrNamespace, n) mv acc -> case mAttrNamespace of
+        Nothing -> do
+          case mv of
+            Just v -> do
+              let [attr'] = toAttr r v
+              return $ " " <> attr'
+            Nothing -> return acc
+        Just ns -> do
+          error $ "modifyAttributes Just " ++ show ns) ""
+
+      -- attributes that were not available till PostBuild was fired
+      -- TODO use map and merge with initial attributes?
+      extraAttrs <- hold "" initial
+      modify (open' extraAttrs <> innerHtml <> close' :)
+      return (Element es (error "Static.element RawElement was used"), result)
+
+  {-# INLINABLE placeholder #-}
+  placeholder (PlaceholderConfig toInsertAbove _delete) = StaticDomBuilderT $ do
+    result <- lift $ performEvent (fmap runStaticDomBuilderT toInsertAbove)
+    acc <- foldDyn (:) [] (fmap snd result)
+    modify . (:) $ join $ fmap (mconcat . reverse) $ current acc
+    return $ Placeholder (fmap fst result) never
+
+  {-# INLINABLE inputElement #-}
+  inputElement cfg = do
+    (e, _result) <- element "input" (cfg ^. inputElementConfig_elementConfig) $ return ()
+    let v0 = constDyn $ cfg ^. inputElementConfig_initialValue
+    let c0 = constDyn $ cfg ^. inputElementConfig_initialChecked
+    let valueChangedByUI = never
+    let hasFocus = constDyn False -- XXX should this coming from initialAtttributes
```

Comment:
And yes, in theory, we should try to get this somehow; but let's leave it for now.